### PR TITLE
feat(security): Containers run as non-root user

### PR DESCRIPTION
**Pull request by beaufrusetta**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/developer-scripts/blob/master/.github/Contributing.md.

## What is the current behavior?
Containers run as root users.

## Issue Number:
#371 

## What is the new behavior?
Containers run as non-root user.

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?
No

## Other information

### CODE REVIEW

**Review by lenny-intel — DISMISSED** [review]
LGTM

**Comment by lenny-intel** [comment]
recheck

**Comment by lenny-intel** [comment]
recheck

**Comment by lenny-intel** [comment]
@cherrycl the failures on this PR look un-related. 
PR https://github.com/edgexfoundry/developer-scripts/pull/373 has similar failures and it should not fail as those changes don't impact the current TAF files. 
Please let us know if there are bugs that need to be addressed to resolve these failures.

**Comment by lenny-intel** [comment]
recheck

**Comment by beaufrusetta** [comment]
Looks like the Redis container runs as "root"...probably not the best idea to have the database image running super privileged. Should I attempt to modify the running user? @lenny-intel

**Comment by lenny-intel** [comment]
@beaufrusetta , yes we should see if those can be run as non-root user.

**Comment by lenny-intel** [comment]
recheck

**Comment by lenny-intel** [comment]
recheck

**Comment by lenny-intel** [comment]
recheck

**Comment by lenny-intel** [comment]
@cherrycl , can you please help us determine the failures on this PR. Logs don't give mush info. THX!

**Comment by cherrycl** [comment]
recheck

**Comment by cherrycl** [comment]
@lenny-intel Sorry for the response lately. I tried to run the tests with this PR's change. It doesn't seem to fail on the same test cases always. Once the failure occurred I found the following error on `redis` and `core-metadata` logs. Hope it can help.
Redis log
```
1:M 14 Jan 2021 11:37:47.077 * 100 changes in 300 seconds. Saving...
1:M 14 Jan 2021 11:37:47.078 * Background saving started by pid 108
108:C 14 Jan 2021 11:37:47.079 # Failed opening the RDB file dump.rdb (in server root dir /data) for saving: Permission denied
1:M 14 Jan 2021 11:37:47.179 # Background saving error
```
core-metadata log
`level=ERROR ts=2021-01-14T11:28:43.466309048Z app=edgex-core-metadata source=deviceservice.go:83 X-Correlation-ID=6a10411d-c9d6-4342-97ab-f7bc93531b99 msg=" -> device service creation failed -> MISCONF Redis is configured to save RDB snapshots, but it is currently not able to persist on disk. Commands that may modify the data set are disabled, because this instance is configured to report errors during writes if RDB snapshotting fails (stop-writes-on-bgsave-error option). Please check the Redis logs for details about the RDB error."`

**Comment by lenny-intel** [comment]
> Once the failure occurred I found the following error on `redis` and `core-metadata` logs. Hope it can help.
> Redis log

@cherrycl , thank you very much!

**Comment by lenny-intel** [comment]
recheck

**Comment by lenny-intel** [comment]
recheck

**Comment by lenny-intel** [comment]
recheck

**Comment by lenny-intel** [comment]
recheck

**Comment by lenny-intel** [comment]
recheck

**Comment by lenny-intel** [comment]
recheck

**Comment by lenny-intel** [comment]
@cherrycl , @jinlinGuan , TAF smoke tests still failing on this PR. They seem unrelated, but they are passing on my PR.
Note clear why they are failing. Can you run them locally against this PR? THX!
https://jenkins.edgexfoundry.org/blue/organizations/jenkins/edgexfoundry%2Fedgex-taf-pipelines/detail/smoke-test/152/tests

**Comment by jinlinGuan** [comment]
@lenny-intel, I suggest we merge PR #373 first, so TAF can get the correct compose files, and that should fix the test failures. 

Because currently TAF generates the compose files that don't include the changes in this PR, for example, `security_opt` and `user` are not set for device-virtual and app services defined in TAF.

**Comment by lenny-intel** [comment]
> I suggest we merge PR #373 first, so TAF can get the correct compose files

OK, we can try that.

**Comment by cloudxxx8** [comment]
we might also need to merge https://github.com/edgexfoundry/edgex-taf/pull/257

**Comment by lenny-intel** [comment]
> we might also need to merge [edgexfoundry/edgex-taf#257](https://github.com/edgexfoundry/edgex-taf/pull/257)

Yes, once we mere #373 we have to merge edgexfoundry/edgex-taf#257

**Comment by lenny-intel** [comment]
@beaufrusetta , I just merged my PR. Please rebase and add `user` to services in all the new partial compose files. THX!

File: compose-builder/add-mqtt-messagebus.yml [hunk 1]
```diff
@@ -19,6 +19,19 @@
...
+      - "1883:1883"
+    container_name: edgex-mqtt-broker
+    hostname: edgex-mqtt-broker
+    read_only: true
+    networks:
+      - edgex-network
+    security_opt: 
+      - no-new-privileges:true
+    user: "${EDGEX_USER}:${EDGEX_GROUP}"
+
```
Comment (jim-wang-intel) [inline compose-builder/add-mqtt-messagebus.yml:34]: This should be removed, right? has been separated out into two sections i guess

**Review by lenny-intel — APPROVED** [review]
LGTM

**Review by lenny-intel — CHANGES_REQUESTED** [review]
Need to run `make build` once more so the new TAF compose files are generated.

**Comment by beaufrusetta** [comment]
> Need to run `make build` once more so the new TAF compose files are generated.

Ok - they have been updated.

**Review by lenny-intel — APPROVED** [review]
LGTM

**Review by jim-wang-intel — APPROVED** [review]

**Pull request merged**